CONTRIBUTING.md

```diff
@@ -65,3 +68,41 @@ using [a phrase such that github auto-closes the issue when merged to master](ht
 [issue-tracker]: https://github.com/CCPPETMR/SIRF/issues
 [pull-requests]: https://github.com/CCPPETMR/SIRF/pulls
 [googlegroup]: https://groups.google.com/forum/#!forum/ccp-petmr-codebot
+
+
+## Submodules
+
+> [...] A submodule is a repository embedded inside another repository. The
+> submodule has its own history [...]
+
+This project uses [submodules] to point to external libraries or frequently
+changed/large datasets.
+
+1. how to pull updates to SIRF
+```bash
+# first time
+SIRF$ git clone https://github.com/CCPPETMR/SIRF --recursive
+# subsequently
+SIRF$ git pull
+SIRF$ git submodule update --init --recursive
+```
+2. how to push updates to the submodule in SIRF
+```bash
+SIRF/data$ git commit -m "add some data" && git push && cd ..
+SIRF$ git add data && git commit -m "update submodule" && git push
+```
+3. what happens when switching branches
+    + submodules not touched when there are conflicts/unstaged changes
+    + have to manually update (which is why `git add --all` is bad practice)
+
+### Caveats
```

Comment:
I find this section a bit confusing. I've suggested a sentence below, but maybe we don't need this section at all?